lib/runtime/src/component.rs

```diff
@@ -255,6 +260,57 @@ impl Component {
             .await
     }
 
+    /// Register Prometheus metrics for this component's service stats
+    pub fn register_metrics_callback(
+        &self,
+    ) -> Result<crate::service::ComponentSystemStatusNatsMetrics> {
+        let component_metrics =
+            crate::service::ComponentSystemStatusNatsMetrics::from_component(self)?;
```

Comment:
question: since this is a component system status object - would new() make sense here instead of from_component() ? i.e. that is it can only be created from a component.